WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -0,0 +1,617 @@
+    }
+    
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        
+        if let page = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? Page {
+            if cellIdentifierForPage(page) == self.dynamicType.restorePageCellIdentifier {
+                return CGFloat(self.dynamicType.pageCellEstimatedRowHeight)
+            }
+            
+            let width = CGRectGetWidth(tableView.bounds)
+            return self.tableView(tableView, heightForRowAtIndexPath: indexPath, forWidth: width)
+        } else {
+            return 0
+        }
+    }
+    
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath, forWidth width: CGFloat) -> CGFloat {
+        configureCell(cellForLayout, atIndexPath: indexPath)
+        let size = cellForLayout.sizeThatFits(CGSizeMake(width, CGFloat.max))
+        let height = ceil(size.height)
+        
+        return height
+    }
+    
+    func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return CGFloat(self.dynamicType.pageSectionHeaderHeight)
+    }
+    
+    func tableView(tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return CGFloat.min
+    }
+    
+    func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView! {
+        let sectionInfo = tableViewHandler?.resultsController.sections?[section]
+        let nibName = String(PageListSectionHeaderView)
+        let headerView = NSBundle.mainBundle().loadNibNamed(nibName, owner: nil, options: nil)[0] as! PageListSectionHeaderView
+        
+        if let sectionInfo = sectionInfo {
+            headerView.setTite(sectionInfo.name)
+        }
+        
+        return headerView
+    }
+    
+    func tableView(tableView: UITableView, viewForFooterInSection section: Int) -> UIView! {
+        return UIView(frame: CGRectZero)
+    }
+    
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        tableView.deselectRowAtIndexPath(indexPath, animated: true)
+        
+        guard let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? AbstractPost
+            where post.remoteStatus != AbstractPostRemoteStatusPushing && post.status != PostStatusTrash else {
+            return
+        }
+        
+        editPage(post)
+    }
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let page = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as! Page
+        
+        let identifier = cellIdentifierForPage(page)
+        let tableViewCell = tableView.dequeueReusableCellWithIdentifier(identifier, forIndexPath: indexPath)
+        
+        precondition(tableViewCell is PageListCell)
+        let cell = tableViewCell as! PageListCell
```

Comment:
Since I need both `tableViewCell` and `cell` after this point (see lines below), it makes sense to keep both steps for now.

You're right though in that this is not elegant.  Solving this is not trivial enough, and I'd rather do that in a follow up PR.

I created [a ticket to improve this code](https://github.com/wordpress-mobile/WordPress-iOS/issues/5236) after this PR is merged.